python/pyspark/sql/tests/streaming/test_streaming.py

```diff
@@ -28,7 +28,7 @@
 
 class StreamingTestsMixin:
     def test_streaming_query_functions_basic(self):
-        df = self.spark.readStream.format("rate").option("rowsPerSecond", 10).load()
+        df = self.spark.readStream.format("text").load("python/test_support/sql/streaming")
```

Comment:
this is just a minor fix it will make the test more stable. We call `query.processAllAvailable` below, which has a very small chance of being a indefinite blocking call on rate source.